### PR TITLE
Add endpoint for getting benefits on a tree-by-tree basis for many trees

### PR DESCRIPTION
**Pull request by maurizi**
Connects to OpenTreeMap/otm-core#2209

### CODE REVIEW

**Comment by maurizi** [comment]
This has some performance issues that need to be addressed, but can perhaps be done in a later PR.

When I attempted to export ecobenefits for all 450,000 trees in LA, the ecoservice ran out of memory and was killed.

I think to tackle this in a satisfactory way will require a refactor to switch to a different serialization method, such as using streaming JSON or ProtoBuffer, which will require switching away from `go-rest`.

File: eco/eco.go [hunk 1]
```diff
@@ -80,7 +84,7 @@ type Fetchable interface {
 // Note that the ith element of the datafiles slice is
 // the ith factor from eco.Factors
 //
-func CalcBenefitsWithData(
+func CalcBenefitSummaryWithData(
```
Comment (steventlamb) [inline eco/eco.go]: `CalcBenefitSummaryWithData` -> `CalcSummarizedBenefitsWithData` would be more consistent with the two other functions it is similar to.

File: ecorest/main.go [hunk 2]
```diff
@@ -8,11 +8,12 @@ import (
...
 type restManager struct {
-	ITreeCodesGET      (func() *endpoints.ITreeCodes)
-	EcoGET             (func(url.Values) (*endpoints.BenefitsWrapper, error))
-	EcoSummaryPOST     (func(*endpoints.SummaryPostData) (*endpoints.BenefitsWrapper, error))
-	EcoScenarioPOST    (func(*endpoints.ScenarioPostData) (*endpoints.Scenario, error))
-	InvalidateCacheGET (func())
+	ITreeCodesGET       (func() *endpoints.ITreeCodes)
+	EcoGET              (func(url.Values) (*endpoints.BenefitsWrapper, error))
+	EcoSummaryPOST      (func(*endpoints.SummaryPostData) (*endpoints.BenefitsWrapper, error))
+	EcoFullBenefitsPOST (func(*endpoints.FullBenefitsPostData) (*endpoints.FullBenefitsWrapper, error))
```
Comment (steventlamb) [inline ecorest/main.go]: I think `EcoListPOST` would be more consistent with `EcoSummaryPOST`, as one provides a list-level result and the other an aggregated result. I think word `Full` would hint more to criteria for inclusion and not output format.

File: ecorest/endpoints/eco_full.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	Instance_id string
+}
+
+// We can't marshall maps directly with
+// go-rest so we just wrap it here
+type FullBenefitsWrapper struct {
+	Benefits map[string]map[string]float64
+}
+
+func EcoFullBenefitsPOST(cache *cache.Cache) func(*FullBenefitsPostData) (*FullBenefitsWrapper, error) {
```
Comment (steventlamb) [inline ecorest/endpoints/eco_full.go]: This file is nearly identical to `eco_summary.go`. Why not have the outer function also take a benefit-calculator function to close over, or, if that makes the signatures inconvenient, introduce another layer of currying or a helper function.

**Pull request closed without merging**